glide-core/src/client/types.rs

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+#[derive(Default)]
+pub struct ConnectionRequest {
+    pub read_from: Option<ReadFrom>,
+    pub client_name: Option<String>,
+    pub authentication_info: Option<AuthenticationInfo>,
+    pub database_id: i64,
+    pub protocol: Option<redis::ProtocolVersion>,
+    pub tls_mode: Option<TlsMode>,
+    pub addresses: Vec<NodeAddress>,
+    pub cluster_mode_enabled: bool,
+    pub request_timeout: Option<u32>,
+    pub connection_retry_strategy: Option<ConnectionRetryStrategy>,
+}
+
+pub struct AuthenticationInfo {
+    pub username: Option<String>,
+    pub password: Option<String>,
```

Comment:
I'm not too familiar with Rust, but password is a required field in the Java client, should this be non-optional?

Reply:
I preferred to keep close to the protobuf implementation. If we want further improvements, we can do them in a separate PR.